### DISCUSSION TITLE
Word Count on nodes

### DISCUSSION THREAD

**Discussion by jandye** [freeplane/freeplane#575] [Help]
I used to have a script that would give me a word count on the selected nodes and/or nested child nodes. Pretty sure I found it in the script library on the previous wiki site, or maybe it was a random forum online somewhere. But for whatever reason I can't find anything similar here, on the wiki or the new help, or even with a google search.

Anyone have something like this they can share? If I recall it was a pretty simple script with just a few lines of code. It would give the word count of entire mindmap, of just the selected nodes, and the selected nodes + child nodes in a dialog box

**Comment by jandye** [comment]
Ok so I found this code from https://www.freeplane.org/forum2012/viewtopic.php?f=1&t=506
```
//@ExecutionModes({ON_SINGLE_NODE})
def selected = c.selecteds
def countWords = selected.sum(0){ it.plainTextContent.split("\\s+").length }
def countChars = selected.sum(0){ it.plainTextContent.length() }
def message = "${countWords} words, ${countChars} characters in ${selected.size()} selected nodes"
ui.informationMessage(ui.frame, message, "Word Count")
```
This might actually be the one I used before, and I was just misremembering what information it included in the dialog box. It only does the word count on the selected nodes, is it possible to include the word count of the selected nodes + any child nodes of the selected nodes?

**Reply by macmarrum** [reply]
Please try this one: [countWordsInSelectedAndDescendants.groovy](https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/countWordsInSelectedAndDescendants.groovy)

**Reply by jandye** [reply]
thanks works perfectly
